packages/firebase_remote_config/example/lib/main.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:async';
+
+import 'package:firebase_remote_config/firebase_remote_config.dart';
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+      new MaterialApp(title: 'Remote Config Example', home: new MyHomePage()));
+}
+
+class MyHomePage extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return new MyHomePageState();
+  }
+}
+
+class MyHomePageState extends State<MyHomePage> {
+  RemoteConfig rc;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Remote Config Example'),
+      ),
+      body: new FutureBuilder<void>(
+          future: setupRemoteConfig(),
+          builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+            switch (snapshot.connectionState) {
+              case ConnectionState.none:
+                return const Center(
+                  child: const Text('loading...'),
+                );
+              case ConnectionState.waiting:
+                return const Center(
+                  child: const Text('waiting...'),
+                );
+              default:
+                if (snapshot.hasError) {
+                  print(snapshot.error);
+                  return const Center(
+                    child: const Text('Failed to show welcome message'),
+                  );
+                } else {
+                  return new Center(
+                    child: new Text('Welcome: ${rc.getString('welcome')}'),
+                  );
+                }
+            }
+          }),
+      floatingActionButton: new FloatingActionButton(
+          child: new Icon(Icons.refresh),
+          onPressed: () {
+            setState(() {});
```

Comment:
This could do all the stuff you have in `setupRemoteConfig` followed by `setState(() {});`

I'm wondering if `RemoteConfig` should be a `ChangeNotifier` so that you can more easily automatically rebuild sections of your widget tree when a new config is activated. But that's easy to add later.

Reply:
Switched to ChangeNotifier